add "permission" parameter to repo.add_collaborator

kuepe-sl opened the pull request:
This PR adds the "permission" parameter to `repo.add_collaborator`, which was missing and is only valid for organisation repositories. (see [here](https://docs.github.com/en/rest/collaborators/collaborators#add-a-repository-collaborator--parameters))

Fixes #954

staticdev reviewed (changes requested):
@kuepe-sl thanks for your PR.

Would you mind checking our [Contributing guide](https://github.com/sigmavirus24/github3.py/blob/main/CONTRIBUTING.rst)? There are some small points that need adjustments.. for instance:

* Passing linters check
* Adding test case for the permission being used

kuepe-sl commented:
Thanks for the hint. I will add a test case.

However I will *not* fix the linter error, because those are caused by the upstream `main` branch and not by me.

staticdev commented:
> Thanks for the hint. I will add a test case.
> 
> However I will _not_ fix the linter error, because those are caused by the upstream `main` branch and not by me.

Thanks, I fixed upstream, if you rebase your PR everything should pass now.

kuepe-sl commented:
Thanks for fixing upstream.  
I rebased my commits and added a unit test.

I didn't add an integration test though, because I'm not sure how to properly execute + record that. It requires access to an Organisation and right now I only have access to private Organisation repos.

kuepe-sl commented:
Thanks a lot for merging!

staticdev commented:
Thank you for the PR.
